app/models/order_item.rb

```diff
@@ -40,10 +40,13 @@ def set_defaults
   end
 
   def update_message(level, message)
-    self.updated_at = DateTime.now
-    ProgressMessage.create(:level         => level,
-                           :message       => message,
-                           :order_item_id => id)
+    time_now = Time.zone.now
```

Comment:
@syncrou I think this should be explicitly set to UTC since these servers could be in different zones and we dont want to use the local time where the server is.

Reply:
@mkanoor since the default time of the app is UTC we are using the default time when using the `zone` method. If `Time.now` was used without `zone` we run into the issue you are talking about.

Reply:
@syncrou So I guess the question is, does `Time.now.utc` protect us from the default timezone of the app changing?  I'm not sure there is much chance of that, but thought I would ask.  I'm good with this either way.

Reply:
Yes.  If we want everything saved in `utc` we either leave the default time zone and call `Time.zone.now` or we call `Time.now.utc` and will be safe from configuration changes such as `config.time_zone = "Pacific Time (US & Canada)"` in `config/application.rb`.

Reply:
We should be letting Rails set the timestamps for us.

Can this be reduced to:
```ruby
progress_messages.create(:level => level, :message    => message)
self.touch
```

Also note that the current code is not associating the progress_message with the order_item.  With the code above that is done for you.

Reply:
@mkanoor - Do the timestamps for `OrderItem` need to be perfectly in sync with the `ProgressMessage` timestamps? If not, I can use `self.touch` and pull out the specific `time_now` entries for the new `ProgressMessage` object.

Reply:
@syncrou The touch is ok